BirtherDay/Presentation/MyCoupons/Interaction/CouponInteractionView.swift

```diff
@@ -6,70 +6,67 @@
 //
 
 import SwiftUI
-
 struct CouponInteractionView: View {
-    var viewModel: CouponDetailViewModel
+    @Bindable var viewModel: CouponDetailViewModel
     
-    @State var distance: Float = 0.8   // TODO: - distance 0.00m 연결
     var minimuDetectedDistance: Float = 2.0
     var screenHeight: CGFloat = UIScreen.main.bounds.height
     
+    @State private var animatedHeight: CGFloat = 0.0
+    @State private var animatedDistance: Float = 0.0
+
     var body: some View {
         ZStack {
             VStack(spacing: 0) {
                 Color.bgDark.ignoresSafeArea()
-                Color.mainPrimary.ignoresSafeArea()
-                    .frame(height: calDistanceToScreenHeight()) // distance에 따른 높이 조절
+                Color.mainPrimary
+                    .ignoresSafeArea()
+                    .frame(height: animatedHeight)
+                    
             }
-            VStack(spacing: 30) {   // TODO: - 쿠폰 영역에 따른 dashed line 크기 수정하기
+
+            VStack(spacing: 30) {
                 VStack {
                     Text("서로의 휴대폰을\n조금 더 가까이 이동시켜주세요!")
                         .font(.sb5)
                         .multilineTextAlignment(.center)
                         .foregroundStyle(Color.bgLight)
                     
+                    BDTemplate(data: viewModel.selectedCoupon, isShowSubtitleView: true, horizontalPadding: 44)
+                        .scaleEffect(204/320)
+                        .rotationEffect(.degrees(7.86))
+
                     Text("0.5m 이내로 가까워졌을 때 사용이 가능해요.")
                         .font(.m1)
                         .foregroundStyle(Color.gray100)
                 }
-//                BDTemplate(data: <#Coupon#>)
-//                    .scaleEffect(204/320)
-//                    .rotationEffect(.degrees(7.86))
-                    
+
                 distanceView()
             }
-                
         }
         .onAppear {
-//            viewModel.startNI()
+            viewModel.startNI()
+        }
+        .animation(.linear(duration: 0.3), value: viewModel.distance)
+        .onChange(of: viewModel.distance ?? 0.0) { _, new in
+            let new = max(new, 0)
+            animatedDistance = new
+            let height = screenHeight - CGFloat(new / minimuDetectedDistance) * screenHeight
+            animatedHeight = max(0, height)
```

Comment:
p3

위 코드는 변수명을 조금 다듬을 필요가 있다는 생각이 들어요!

let new = max(new, 0)으로 할 시 다른 개발자가 코드를 볼때 어떤 값으로 클램핑 되어 사용되는지 한번에 파악하기 어려울 것 같다는 생각이 들었어용!

함수형 프로그래밍의 개념 중 불변성(immutable)을 지킨다면 유지보수, 가독성 측면에서 장점을 누릴 수 있어서 인자(_, new)의 값은 수정하지 않는 것이 좋아보여요!

```suggestion
let clampedDistance = max(new, 0)
animatedDistance = clampedDistance

let progress = CGFloat(clampedDistance / minimuDetectedDistance)
let adjustedHeight = screenHeight * (1 - progress)
animatedHeight = max(0, adjustedHeight)
```